### ISSUE TITLE
Syntax for metadata on code blocks

### ISSUE THREAD

**Issue by Julow** [ocaml/odoc#633]
It should be possible to attach metadata to code blocks. There is two main use-cases:

- Running Mdx on OCaml and mld files. It needs per-block configuration.
- Code highlighting. The HTML backend adds color highlighting but assumes that all blocks contain OCaml code.

Odoc won't try to interpret or to render them, they would be present in the parse tree but discared afterwards.

During a previous meeting, we proposed the following syntax:
```
{@language metadata [
]}
```

The `metadata` part is an arbitrary string and is optional. 
We'd need quoting, I propose wrapping it with double quotes (`""`) and allowing OCaml's escaping (eg. `\"` and everything `String.escape` produces).

The `language` part is not optional.

For example:
```
{@sh [
rm -rf /
]}
```
```
{@ocaml env=f1 version>=4.06 [
# let x = 1 ;;
val x : 1
]}
```

**Comment by dbuenzli** [comment]
Code extraction is another use case. There's a long discussion about it in https://github.com/ocaml/odoc/pull/303

**Comment by jonludlam** [comment]
@dbuenzli yes, I'm hoping mdx will do that job for us. It's got code extraction working already (but from markdown files).

**Comment by dbuenzli** [comment]
Honestly I rather not add more tools to the basics I need for dev (and especially not tools that do [black magic](https://github.com/realworldocaml/mdx/blob/931dc946877ddc220964f29b34ad6de0516a10e0/lib/top/mdx_top.ml#L535)). It doesn't seem super hard to add file extraction to `odoc` itself by starting from the document IR.

**Comment by Julow** [comment]
This is implemented in https://github.com/ocaml-doc/odoc-parser and has been released (1.0.0).
